### PR TITLE
[DEVX-1081] Update for playwright 1.12

### PR DESCRIPTION
**Pull request by mhan83**
Bumping playwright to 1.12.3 which has a new api for playwright-test so updating example tests as well.

### CODE REVIEW

File: .sauce/config.yml [hunk 1]
```diff
@@ -30,6 +30,16 @@ suites:
     testMatch: '**/*.js'
     params:
       browserName: "chromium"
+  - name: "Firefox in docker"
+    mode: docker
+    testMatch: '**/*.js'
```
Comment (FriggaHel) [inline .sauce/config.yml:35]: That needs to be regex now:

```suggestion
    testMatch: '.*.js'
```

(Error was not seen due to https://github.com/saucelabs/sauce-playwright-runner/pull/106)

File: .sauce/config.yml [hunk 2]
```diff
@@ -30,6 +30,16 @@ suites:
...
+    params:
+      browserName: "firefox"
+  - name: "Webkit in docker"
+    mode: docker
+    testMatch: '**/*.js'
```
Comment (FriggaHel) [inline .sauce/config.yml:40]: That needs to be regex now:

```suggestion
    testMatch: '.*.js'
```

(Error was not seen due to https://github.com/saucelabs/sauce-playwright-runner/pull/106)

File: .sauce/config.yml [hunk 3]
```diff
@@ -16,30 +16,40 @@ docker:
...
 playwright:
-  version: 1.11.1 # See https://docs.saucelabs.com/testrunner-toolkit for a list of supported versions.
+  version: 1.12.3 # See https://docs.saucelabs.com/testrunner-toolkit for a list of supported versions.
 # Controls what files are available in the context of a test run (unless explicitly excluded by .sauceignore).
 rootDir: ./
 suites:
   - name: "Firefox using global mode" # Since the suite doesn't specify the `mode`, it'll inherit the mode specified via `defaults.mode` (see line number 3 and 4 of this config file).
     platformName: "Windows 10" # Only relevant when running a test against the sauce cloud.
-    testMatch: '**/*.js'
+    testMatch: '.*.js'
```
Comment (alexplischke) [inline .sauce/config.yml:25]: Let's change this to an array, as that's the structure we official expect 💪 
```suggestion
    testMatch: ['.*.js']
```

Comment (alexplischke) [reply .sauce/config.yml:25]: PS: Since this is now a regex, I think this might also pick up any tests nested in the `examples/` folder.

**Comment by mhan83** [comment]
I also removed the compile step for the typescript example since playwright handles it natively. I'm not totally sure we need the typescript example, tbh.

**Review by tianfeng92 — APPROVED** [review]

File: examples/typescript/.sauce/config.yml [hunk 4]
```diff
@@ -16,19 +16,19 @@ docker:
...
 playwright:
-  version: 1.11.1 # See https://docs.saucelabs.com/testrunner-toolkit for a list of supported versions.
+  version: 1.12.3 # See https://docs.saucelabs.com/testrunner-toolkit for a list of supported versions.
 # Controls what files are available in the context of a test run (unless explicitly excluded by .sauceignore).
 rootDir: ./
 suites:
   - name: "Chromium in docker"
     mode: docker
-    testMatch: '**/*.js'
+    testMatch: '.*.ts'
```
Comment (FriggaHel) [inline examples/typescript/.sauce/config.yml:25]:
```suggestion
    testMatch: ['.*.ts']
```

File: examples/typescript/.sauce/config.yml [hunk 5]
```diff
@@ -16,19 +16,19 @@ docker:
...
     params:
       browserName: "chromium"
   - name: "Chromium in sauce"
     mode: sauce
     platformName: "Windows 10"
-    testMatch: '**/*.js'
+    testMatch: '.*.ts'
```
Comment (FriggaHel) [inline examples/typescript/.sauce/config.yml:31]:
```suggestion
    testMatch: ['.*.ts']
```

**Review by FriggaHel — APPROVED** [review]

**Pull request merged**